builds/jazz-build-module/lambda-events-module.groovy

```diff
@@ -0,0 +1,175 @@
+  try {
+    if (existing_notifications.containsKey("LambdaFunctionConfigurations")) {
+      def existing_lambda_configs = existing_notifications.LambdaFunctionConfigurations
+      new_events = getLambdaEvents(existing_notifications, events)
+      if (new_events != null && new_events.size() > 0) {
+        for(item in existing_lambda_configs){
+          lambdaFunctionConfigurations.add(item)
+        }
+        new_lambda_configuration.Events = new_events
+        lambdaFunctionConfigurations.add(new_lambda_configuration)
+      }
+    } else {
+      new_events = checkAndConvertEvents(events)
+      new_lambda_configuration.Events = new_events
+      lambdaFunctionConfigurations.add(new_lambda_configuration)
+    }
+    if (lambdaFunctionConfigurations != null && lambdaFunctionConfigurations.size() > 0) {
+      existing_notifications.LambdaFunctionConfigurations = lambdaFunctionConfigurations
+      def newNotificationJson = JsonOutput.toJson(existing_notifications)
+      def response = sh(returnStdout: true, script: "aws s3api put-bucket-notification-configuration --bucket $s3BucketName --notification-configuration \'${newNotificationJson}\' --output json")
+    }
+  } catch (ex) {
+  }
+}
+
+def getLambdaEvents(existing_notifications, events){
+  def existing_events = []
+  def new_events = []
+  for (item in events){
+      new_events.add(item)
+    }
+  for(item in existing_notifications.LambdaFunctionConfigurations ){
+    existing_events.addAll(item.Events)
+  }
+  def cleanupIndex = -1
+  echo "events . $events"
+
+ // Removing the existing events from the new event list
+ def event_names
+  for(item in events){
+    cleanupIndex++
+    if(( item.contains("ObjectCreated") || item.contains("ObjectRemoved" )) &&
+     ( existing_events.contains("s3:ObjectCreated:*")  ||  existing_events.contains("s3:ObjectRemoved:*") )){
+        new_events[cleanupIndex] = null
+    } else if (existing_events.contains(item)){
+        new_events[cleanupIndex] = null
+    }
+  }
+  new_events.removeAll([null])
+  def events_list = []
+  if(new_events.size() > 0 && new_events != null){
+    events_list = checkAndConvertEvents(new_events)
+  }
+  return events_list
+}
+
+def checkAndConvertEvents(events){
+  def new_events = []
+  // converting the new events to * event
+  def cleanupIndex = -1
+  def isCreationEvent = false
+  def isRemovalEvent = false
+  if(events.size() > 0 && events != null){
+    for (item in events){
+      new_events.add(item)
+    }
+
+    for(item in events){
+      cleanupIndex++
+      if(item.contains("ObjectCreated")) {
+        isCreationEvent = true
+        new_events[cleanupIndex] = null
+      }else if(item.contains("ObjectRemoved" )) {
+        isRemovalEvent = true
+        new_events[cleanupIndex] = null
+      }
+    }
+
+    new_events.removeAll([null])
+    if(isCreationEvent == true){
+      new_events.add("s3:ObjectCreated:*")
+    }
+    if(isRemovalEvent == true){
+      new_events.add("s3:ObjectRemoved:*")
```

Comment:
I tried this and it seems you cannot have both ObjectCreated and ObjectRemoved. I got this error
Configurations on the same bucket cannot share a common event type. (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: C156EB32EE8CB59D; S3 Extended Request ID: 

So you can remove all of this.

Reply:
While I am trying I was able to put the notifications. Please give me the test events and the scenario you had tried. So that I can give the fix